### DISCUSSION TITLE
Building column level lineage

### DISCUSSION THREAD

**Discussion by igormasternoy** [reata/sqllineage#473] [Q&A]
I'm having a following use-case, where I'd be glad to use this amazing tool. 
We have some queries in Tableau, which are taking data from Vertica. Queries usually are just DMLs (some with subqueries and resulting select) meaning there is no insert into statements. We would like to build a column level lineage from Vertica tables to this Query. To understand what are actual columns are being used in Tableau.
As far as I understood to achieve the column level lineage I just need to add an insert into foo in the end of the DML query.

**Comment by reata** [comment]
Yes, SELECT only SQLs won't present column lineage for current implementation. Appending an INSERT INTO FOO to the end does work, however it's relying on a buggy feature that we don't validate your SQL for default dialect.

The more reasonable thing to do is prepend `INSERT INTO FOO` at the beginning so it would be valid syntax and won't break in future release when we switch to a default validating mode (and we will in v1.5.x).

Personally if I'm doing this, I would prepend `CREATE VIEW dashboard_xxx AS` at the beginning instead, because what Tableau does to me is quite like creating a view though not persisted in database.

Hope this helps.
